core/src/main/scala/org/apache/spark/sql/delta/commands/RestoreTableCommand.scala

```diff
@@ -124,7 +119,6 @@ case class RestoreTableCommand(
             metrics)
         } finally {
           filesToAdd.unpersist()
```

Comment:
Also did you notice any perf differences with or without the `cache`?

Reply:
@vkorukanti Good point. I've moved decision login (validate or not) to `run` method and added check before unpersist.

Regarding performance - I don't have the answer, because I don't have access to really big delta tables. But for tables with a lot of files I think it will be useful

Reply:
@vkorukanti I've tested performance with and without cache on table with 50k files. Full table restore (50k add actions and 50k remove actions) without cache is 2x faster than with cached dataframes. Looks like spark spends a lot of time for caching and much faster to recreate dataframes from the scratch.
So, we don't need cache at all.